src/js/components/locations-configuration/ZoneAndBinLocations.jsx

```diff
@@ -1,26 +1,126 @@
   constructor(props) {
     super(props);
+    this.setShowAddZoneModal = this.setShowAddZoneModal.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+    this.addLocation = this.addLocation.bind(this);
+    this.setShowEditZoneModal = this.setShowEditZoneModal.bind(this);
+    this.handleZoneEdit = this.handleZoneEdit.bind(this);
+    this.editLocation = this.editLocation.bind(this);
+    this.deleteLocation = this.deleteLocation.bind(this);
     this.state = {
+      ...INITIAL_STATE,
+      showAddZoneModal: false,
+      showEditZoneModal: false,
+      valuesToEdit: {},
       values: this.props.initialValues,
     };
   }
 
+
+  setShowAddZoneModal() {
+    this.setState({ showAddZoneModal: !this.state.showAddZoneModal });
+  }
+
+  setShowEditZoneModal() {
+    this.setState({ showEditZoneModal: !this.state.showEditZoneModal });
+  }
+
+  handleZoneEdit(values) {
+    this.setState({
+      showEditZoneModal: !this.state.showEditZoneModal,
+      valuesToEdit: { ...values },
+    });
+  }
+
+  addLocation(data, entity) {
+    if (entity === ZONE) {
+      this.setState({ zoneData: [...this.state.zoneData, data] });
+      return;
+    }
+    // below is prepared for bin location. It will be needed to replace zoneData with binData.
+    this.setState({ zoneData: [...this.state.zoneData, data] });
+  }
+
+  editLocation(editedLocation, entity) {
+    if (entity === ZONE) {
+      this.setState({
+        zoneData: this.state.zoneData.map((location) => {
+          if (location.id === editedLocation.id) {
+            return editedLocation;
+          }
+          return location;
+        }),
+      });
+    }
+    // below will be added setState for bin location, that's why I used that if to check entity
+  }
+
+  deleteLocation(locationId) {
+    confirmAlert({
+      title: this.props.translate('react.locationsConfiguration.deleteZoneConfirm.title.label', 'Deleting a location'),
+      message: this.props.translate(
+        'react.locationsConfiguration.deleteZoneConfirm.subtitle.label',
+        'If you press \'Yes\', this will delete the location. If you decide not to delete the location, press \'No\'',
+      ),
+      buttons: [
+        {
+          label: this.props.translate('react.default.yes.label', 'Yes'),
+          onClick: () => {
+            apiClient.delete(`/openboxes/api/locations/${locationId}`)
+              .then((res) => {
+                if (res.status === 200) {
+                  this.setState({
+                    zoneData: this.state.zoneData.filter(location => location.id !== locationId),
+                  });
+                }
+              })
+              .catch(() => Promise.reject(new Error(this.props.translate('react.locationsConfiguration.editZone.error.label', 'Could not edit zone location'))));
+          },
+        },
+        {
+          label: this.props.translate('react.default.no.label', 'No'),
+        },
+      ],
+    });
+  }
+
+  fetchData(data, entity) {
```

Comment:
I think it will be best to create a separate method instead of using the entity parameter, also it's not used to fetch the data, so you should rename it, it can be updateZoneData